src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -178,19 +178,19 @@ async def _git_clean_repo(directory: str):
 
 async def _git_checkout_files(directory: str, paths: list[Path], tag: Optional[str]):
     if not tag:
-        tag = "HEAD"
-    cmd = ["git", "checkout", tag] + [f"{path}" for path in paths]
+        tag: Literal["HEAD"] = "HEAD"
```

Comment:
TIP: re-annotating a variable is not a very good practice.  An alternative implementation
``python
async def _git_checkout_files(directory: str, paths: list[Path], tag: Optional[str]):tag = tag or "HEAD"
    tag = tag or "HEAD"
    # or 
   tag = "HEAD" if tag is None else tag
``

Reply:
will change this, thx